app/services/middleware_topology_service.rb

```diff
@@ -1,30 +1,27 @@
 class MiddlewareTopologyService < TopologyService
+  include UiServiceMixin
```

Comment:
yes, but this change should be part of that commit. because once the tests are fixed here, since the PR is not marked as wip, it might be merged as is.